static/app/components/smartSearchBar/index.tsx

```diff
@@ -198,6 +198,10 @@ type Props = WithRouterProps &
      * Disables wildcard searches (in freeText and in the value of key:value searches mode)
      */
     disallowWildcard?: boolean;
+    /**
+     * Custom message that will be displayed when the prop `disallowWildcard` is true and the user types a wildcard
+     */
+    disallowWildcardReason?: string;
```

Comment:
Let's implement this a little differently.

I like the [`InvalidReason`](https://github.com/getsentry/sentry/pull/49670/files#diff-5ab58ceffe218bd4a21b4df4aae9ac1e7fb139acb5ff082d0d5e536d38d6b3a5R257-R259) that you have in https://github.com/getsentry/sentry/pull/49670

What I think we should do here is, let's expand the invalid type enum you have to include ALL invalid types.

Then what we can do is instead of having a config here called `disallowWildcardReason`, let's have a prop called `invalidMessages: Record<InvalidReason, string>` and the we can default those to all the messages we have right now, so something like

```tsx
enum InvalidReason {
	MustHaveValue = 'must_have_value',
  MustBeQuoted = 'must_be_quoted',
  InvalidNumber = 'invalid_number',
  InvalidSize = '...',
  InvalidBoolean = '...',
  // There are mroe I didn't include them all
  // ...
  NoWildcard = 'no_wildcard'
}

// And in our `SearchConfig` we add the type
type SearchConfig = {
  // ...
  /**
   * Configures the associated messages for invalid reasons
   */
  invalidMessages: Record<InvalidReason, string>
  // ...
}

// And in our `defaultConfig` we have something like
const defaultConfig = {
  // ...
	invalidMessages: {
    [InvalidReason.MustHaveValue]: t('Filter must have a value'),
    [InvalidReason.MustBeQuoted]: t('Quotes must enclose text or be escaped')};
    [InvalidReason.InvalidNumber]: t('Invalid number. Expected number then optional k, m, or b suffix (e.g. 500k)'),
    [InvalidReason.InvalidSize]: t('...'),
    [InvalidReason.InvalidBoolean]: t('...'),
    // There are mroe I didn't include them all
    // ...
    [InvalidReason.NoWildcard]: t('Wildcards not supported in search'),
  }
  // ...
}
```

Now when we override the `invalidMessages` in with the `additionalConfig` parameter of `parseSearch` we can provide our own custom message, and our parser doesn't have one off code specific to this wildcard stuff, instead it's nicely generalized so if we want to adjust messages for other invalid types we can as well.